concordium-consensus/src/Concordium/GlobalState/Wasm.hs

```diff
@@ -39,96 +32,32 @@ import Data.Serialize
-import Foreign.Ptr
-import Foreign.C
-
-import Concordium.Crypto.FFIHelpers (fromBytesHelper, toBytesHelper)
 import Concordium.Utils.Serialization
 import Concordium.Types
 import Concordium.Wasm
 
-foreign import ccall unsafe "&artifact_v0_free" freeArtifactV0 :: FunPtr (Ptr ModuleArtifactV0 -> IO ())
-foreign import ccall unsafe "artifact_v0_to_bytes" toBytesArtifactV0 :: Ptr ModuleArtifactV0 -> Ptr CSize -> IO (Ptr Word8)
-foreign import ccall unsafe "artifact_v0_from_bytes" fromBytesArtifactV0 :: Ptr Word8 -> CSize -> IO (Ptr ModuleArtifactV0)
-
-foreign import ccall unsafe "&artifact_v1_free" freeArtifactV1 :: FunPtr (Ptr ModuleArtifactV1 -> IO ())
-foreign import ccall unsafe "artifact_v1_to_bytes" toBytesArtifactV1 :: Ptr ModuleArtifactV1 -> Ptr CSize -> IO (Ptr Word8)
-foreign import ccall unsafe "artifact_v1_from_bytes" fromBytesArtifactV1 :: Ptr Word8 -> CSize -> IO (Ptr ModuleArtifactV1)
-
--- | A processed module artifact ready for execution. The actual module is
--- allocated and stored on the Rust heap, in a reference counted pointer.
-newtype ModuleArtifact (v :: WasmVersion) = ModuleArtifact { maArtifact :: ForeignPtr (ModuleArtifact v) }
-  deriving(Eq, Show) -- the Eq and Show instances are only for debugging and compare and show pointers.
-
-type ModuleArtifactV0 = ModuleArtifact V0
-type ModuleArtifactV1 = ModuleArtifact V1
-
--- |Wrap the pointer to the module artifact together with a finalizer that will
--- deallocate it when the module is no longer used.
-newModuleArtifactV0 :: Ptr ModuleArtifactV0 -> IO ModuleArtifactV0
-newModuleArtifactV0 p = do
-  maArtifact <- newForeignPtr freeArtifactV0 p
-  return ModuleArtifact{..}
-
--- |Wrap the pointer to the module artifact together with a finalizer that will
--- deallocate it when the module is no longer used.
-newModuleArtifactV1 :: Ptr ModuleArtifactV1 -> IO ModuleArtifactV1
-newModuleArtifactV1 p = do
-  maArtifact <- newForeignPtr freeArtifactV1 p
-  return ModuleArtifact{..}
-
--- |Use the module artifact temporarily. The pointer must not be leaked from the
--- computation.
-withModuleArtifact :: ModuleArtifact v -> (Ptr (ModuleArtifact v) -> IO a) -> IO a
-withModuleArtifact ModuleArtifact{..} = withForeignPtr maArtifact
-
--- |Load a module artifact from a 'BS.ByteString'. This creates a reference counted copy of
--- the artifact on the Rust side and attaches a finalizer to drop the reference when it is
--- garbage collected on the Haskell side.
-moduleArtifactFromBytes :: forall v. (IsWasmVersion v) => BS.ByteString -> IO (ModuleArtifact v)
-moduleArtifactFromBytes bs = do
-  let martifact = case getWasmVersion @v of
-        SV0 -> fromBytesHelper freeArtifactV0 fromBytesArtifactV0 bs
-        SV1 -> fromBytesHelper freeArtifactV1 fromBytesArtifactV1 bs
-  case martifact of
-    Nothing -> error "Cannot decode module artifact."
-    Just maArtifact -> return ModuleArtifact{..}
+-- |A processed module artifact as a 'BS.ByteString'.
```

Comment:
Please elaborate a bit more here about the provenance, and when this is meant to be used.